core/src/Streamly/Internal/Data/Fold.hs

```diff
@@ -516,82 +575,110 @@ scan = scanWith False
+--
+-- See "Streamly.Internal.Data.Fold.Window".
+--
+{-# INLINE slide2 #-}
+slide2 :: Monad m => Fold m (a, Maybe a) b -> Fold m a b
+slide2 (Fold step1 initial1 extract1) = Fold step initial extract
 
-    extract = extractR . snd
+    where
 
-------------------------------------------------------------------------------
--- Filters
-------------------------------------------------------------------------------
+    initial =
+        first (Tuple' Nothing) <$> initial1
+
+    step (Tuple' prev s) cur =
+        first (Tuple' (Just cur)) <$> step1 s (cur, prev)
+
+    extract (Tuple' _ s) = extract1 s
 
--- | Convert a predicate into a filtering fold.
+-- | Drop repeated elements that are adjacent to each other using the supplied
+-- comparison function.
 --
--- >>> f = Fold.foldFilter (Fold.satisfy (> 5)) Fold.sum
--- >>> Stream.fold f $ Stream.fromList [1..10]
--- 40
+-- To strip duplicate path separators:
+--
+-- >>> input = Stream.unfold Unfold.fromList "//a//b"
+-- >>> f x y = x == '/' && y == '/'
+-- >>> Stream.fold Fold.toList $ Stream.scanMaybe (Fold.uniqBy f) input
+-- "/a/b"
+--
+-- Space: @O(1)@
+--
+-- See also: 'nubBy'.
 --
 -- /Pre-release/
-{-# INLINE satisfy #-}
-satisfy :: Monad m => (a -> Bool) -> Fold m a (Maybe a)
-satisfy f = Fold step (return $ Partial ()) (const (return Nothing))
+--
+{-# INLINE uniqBy #-}
+uniqBy :: Monad m => (a -> a -> Bool) -> Fold m a (Maybe a)
+uniqBy eq = rollingMap f
 
     where
 
-    step () a = return $ Done $ if f a then Just a else Nothing
+    f pre curr =
+        case pre of
+            Nothing -> Just curr
+            Just x -> if x `eq` curr then Nothing else Just curr
 
--- | Use a 'Maybe' returning fold as a filtering scan.
+-- | Drop repeated elements that are adjacent to each other.
 --
--- >>> f = Fold.foldFilter (Fold.satisfy (> 5)) Fold.sum
--- >>> Stream.fold f $ Stream.fromList [1..10]
--- 40
+-- >>> uniq = Fold.uniqBy (==)
 --
--- The above snippet is equivalent to:
+{-# INLINE uniq #-}
+uniq :: (Monad m, Eq a) => Fold m a (Maybe a)
+uniq = uniqBy (==)
+
+-- | Strip all leading and trailing occurrences of an element passing a
+-- predicate and make all other consecutive occurrences uniq.
 --
--- >>> f = Fold.filter (> 5) Fold.sum
--- >>> Stream.fold f $ Stream.fromList [1..10]
--- 40
+-- >> prune p = Stream.dropWhileAround p $ Stream.uniqBy (x y -> p x && p y)
 --
--- /Pre-release/
-{-# INLINE foldFilter #-}
-foldFilter :: Monad m => Fold m a (Maybe b) -> Fold m b c -> Fold m a c
-foldFilter f1 f2 = many f1 (catMaybes f2)
+-- @
+-- > Stream.prune isSpace (Stream.fromList "  hello      world!   ")
+-- "hello world!"
+--
+-- @
+--
+-- Space: @O(1)@
+--
+-- /Unimplemented/
+{-# INLINE prune #-}
+prune ::
+    -- (Monad m, Eq a) =>
+    (a -> Bool) -> Fold m a (Maybe a)
+prune = error "Not implemented yet!"
+
+-- | Emit only repeated elements, once.
+--
+-- /Unimplemented/
+repeated :: -- (Monad m, Eq a) =>
+    Fold m a (Maybe a)
+repeated = undefined
```

Comment:
Any special thing? `error "Not implemented yet!"` vs `undefined`?

Reply:
This was existing code, not changed in this PR.